### PR TITLE
Feat/layout

### PR DESCRIPTION
**Pull request by akshitarora921**
This PR closes [#4057](https://github.com/amplication/amplication/issues/4057)
- add `Layout.js`
- remove Header, Footer and Background components from the pages
- consume layout using per-page Layout method [Reference](https://nextjs.org/docs/basic-features/layouts#per-page-layouts)

### CODE REVIEW

**Comment by yuval-hazaz** [comment]
@MichaelSolati can you please review this PR

File: layouts/index.js [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+import PropTypes from "prop-types";
+
+const Layout = ({
+  children,
+  hideBackground,
+  hideFooterBanner,
+  footerCompactView,
+}) => {
+  return (
+    <div className="page min-h-screen flex flex-col justify-start justify-items-stretch overflow-hidden pt-[60px] laptop:pt-[110px] bg-purple-dark relative">
```
Comment (MichaelSolati) [inline layouts/index.js:13]: This is not working as it used to. On the site right now, this provides padding-top of either 65px or 110px; in your PR that padding is missing.

**Comment by MichaelSolati** [comment]
Things code-wise looked good, but I'm seeing a small issue. The padding from the wrapper div of the header, main, and footer is missing, so if you check out pages like the blog page, you'll notice that the tags filter is missing.

Additionally, while I like that the main layout is just called layout, the layouts folder implies that there should be multiple layouts possible. I'd appreciate it if we rename `layouts/index.js` to `layouts/main.js`, and rename the component from `Layout` to `MainLayout`.

You can use `layouts/index.js` as a file that exports the layout, that way in the code you just import `layouts` and have access to all (one) of the layouts we have.

File: layouts/index.js (hunk 1 above)
Comment (akshitarora921) [reply layouts/index.js:13]: Yes, we are using extra padding top in `index.js`. For index.js it is `143px` and for other pages it is `110px`.
So we have to pass the padding top value to the Layout component.

**Comment by akshitarora921** [comment]
- [x] rename `index.js` to `main.js`
- [x] exporting layouts from `layouts/index.js`
- [x] rename `Layout` to `MainLayout`
- [x] fix padding for `pages/index.js`

Can we add `prettierrc` file as I see there is an inconsistency in code formatting? If yes, Can I work on the same?

**Comment by yuval-hazaz** [comment]
@MichaelSolati please review again

@akshitarora921 yes, please open an issue about the missing `prettierrc` and work on the issue in a separate PR

**Review by MichaelSolati — APPROVED** [review]

**Pull request merged**